bundles/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/internal/handler/PigpioDigitalOutputHandler.java

```diff
@@ -52,34 +63,182 @@ public class PigpioDigitalOutputHandler implements ChannelHandler {
      * @param updateStatus Is called when the state should be changed
      * 
      * @throws PigpioException Can be thrown by Pigpio
-     * @throws NoGpioIdException Is thrown when no gpioId is defined
+     * @throws ChannelConfigurationException Thrown on configuration error
      */
-    public PigpioDigitalOutputHandler(GPIOOutputConfiguration configuration, JPigpio jPigpio,
-            Consumer<State> updateStatus) throws PigpioException, NoGpioIdException {
+    public PigpioDigitalOutputHandler(GPIOOutputConfiguration configuration, ScheduledExecutorService scheduler,
+            Consumer<State> updateStatus) throws PigpioException, ChannelConfigurationException {
         this.configuration = configuration;
+        this.gpioId = configuration.gpioId;
+        this.scheduler = scheduler;
         this.updateStatus = updateStatus;
-        Integer gpioId = configuration.gpioId;
-        if (gpioId == null) {
-            throw new NoGpioIdException();
+
+        if (this.gpioId <= 0) {
+            throw new ChannelConfigurationException("Invalid gpioId value.");
+        }
+
+        if (configuration.pulse.compareTo(BigDecimal.ZERO) > 0) {
+            try {
+                this.pulseTimeout = configuration.pulse.intValue();
+            } catch (Exception e) {
+                throw new ChannelConfigurationException("Invalid expire value.");
+            }
+        }
+
+        if (configuration.pulseCommand.length() > 0) {
+            this.pulseCommand = configuration.pulseCommand.toUpperCase();
+            if (!"ON".equals(pulseCommand) && !"OFF".equals(pulseCommand) && !"BLINK".equals(pulseCommand)) {
+                throw new ChannelConfigurationException("Invalid pulseCommand value.");
+            }
         }
-        this.gpio = new GPIO(jPigpio, gpioId, JPigpio.PI_OUTPUT);
     }
 
+    /**
+     * Future to track pulse commands.
+     */
+    @Nullable
+    private Future<?> pulseJob = null;
+    @Nullable
+    private OnOffType lastPulseCommand;
+
+    /**
+     * Used to only keep a single gpio command handle in flight
+     * at a time.
+     */
+    private Object handleLock = new Object();
+
     @Override
-    public void handleCommand(Command command) {
-        if (command instanceof RefreshType) {
-            try {
-                updateStatus.accept(OnOffType.from(configuration.invert != gpio.getValue()));
-            } catch (PigpioException e) {
-                logger.warn("Unknown pigpio exception while handling Refresh", e);
+    public void handleCommand(Command command) throws PigpioException {
+        synchronized (handleLock) {
+            GPIO lgpio = this.gpio;
+            Consumer<State> lupdateStatus = this.updateStatus;
+            Future<?> job = this.pulseJob;
+
+            if (lgpio == null || lupdateStatus == null) {
+                logger.warn("An attempt to submit a command was made when the pigpiod was offline: {}",
```

Comment:
Not 100% sure, but i think debug would be better.

Reply:
I think something above debug should be warranted here. The reason I had it at warn is due to the fact that this scenario represents a lost command. An action that the user requested could not be carried out. With debug, this would be normally suppressed and not known to the user.  Maybe info?

Reply:
Warm is fine, see other comment. Thanks